src/ui/ui_sign_transaction.c

```diff
@@ -670,8 +677,120 @@ UX_DEF(ux_burn_mint_flow, &summary_step, &operator_step, &senders_step,
 UX_DEF(ux_associate_flow, &summary_step, &operator_step, &senders_step,
        &fee_step, &memo_step, &confirm_step, &reject_step);
 
+
+#elif defined(HAVE_NBGL)
+
+
+static void rejectChoice(void) {
+    io_exchange_with_code(EXCEPTION_USER_REJECTED, 0);
+    nbgl_useCaseStatus("message\nrejected", false, ui_idle);
+}
+
+static void rejectUseCaseChoice(void) {
+    nbgl_useCaseConfirm("Reject message?", NULL, "Yes, reject", "Go back to message", rejectChoice);
+}
+
+static void review_final_callback(bool confirmed) {
+    if (confirmed) {
+        io_exchange_with_code(EXCEPTION_OK, 64);
+        nbgl_useCaseStatus("MESSAGE\nSIGNED", true, ui_idle);
+    } else {
+        rejectUseCaseChoice();
+    }
+}
+
+// Max is 8 infos for transfer transaction
+static nbgl_layoutTagValue_t infos[8];
```

Comment:
I'm a bit afraid of a buffer overflow that might occurs if at some point some the comment below is not true anymore.
Can you add a check below somehow? Or would it get to clumsy?

Reply:
Can't really add a check as this function does not handle errors ...
I agree on the danger of this, I'll add comments